invenio_rdm_records/services/pids/service.py

```diff
@@ -19,7 +19,19 @@ class PIDsService(RecordService):
     def __init__(self, config, manager_cls):
         """Constructor for RecordService."""
         super().__init__(config)
-        self._manager = manager_cls(self.config.pids_providers)
+        self.manager_cls = manager_cls
+
+    @property
+    def _manager(self):
+        """Transitive manager property.
+
+        This is done to:
+        - only access self.config attributes when in an application context
+        - limit code change
+        - limit side-effects (in case using pre-existing `pid_manager` would
+                              cause some.)
+        """
+        return self.manager_cls(self.config.pids_providers)
```

Comment:
That's right. The way it worked before, the constructor was setting the manager (making that `manager_cls(self.config.pids_providers)`) with a passed `config` that had just been customized in `ext.py`. Since we moved that customization out and it does the work of connecting to the `app.config`, we need to delay the creation of the manager (the `_manager` property) until the service is used i.e. until the app context is pushed which is always the case when a service is used (but not when it's created).